### PR TITLE
Fix `fb_season_team_stats()`

### LINKED ISSUE #389: `fb_season_team_stats()` returning error when `stat_type = "league_table"`
I have just started receiving the following error when trying to get league table data from FBref using `fb_season_team_stats()`:

``` r

league_table_raw <- 
  worldfootballR::fb_season_team_stats(
    country = c("ENG", "ESP", "GER", "ITA", "FRA"),
    gender = "M",
    season_end_year = c(2013:2024),
    tier = "1st",
    stat_type = "league_table",
    time_pause = 5
  )
#> Warning in min(league_tables_idx): no non-missing arguments to min; returning
#> Inf
#> Error in `map()`:
#> ℹ In index: 1.
#> Caused by error in `.f()`:
#> ! object 'stat_df' not found
```

This is code that I have previously run multiple times, without issues, though I probably haven't run this code for a few weeks (it's for a blog post I haven't had a chance to work on recently, until today).

I tested a few different values for `season_end_year`, including single values and ranges of values, to ensure it isn't specific to certain seasons, but all efforts produce the same error.

``` r
packageVersion("worldfootballR")
#> [1] '0.6.5.6'
```

### PR DESCRIPTION
**Pull request by tonyelhabr**
Fixes #389

### CODE REVIEW

File: R/get_season_team_stats.R [hunk 1]
```diff
@@ -44,30 +44,39 @@ fb_season_team_stats <- function(country, gender, season_end_year, tier, stat_ty
...
-    season_stats_page <- .load_page(season_url)
+    season_stats_page <- .load_page(single_season_url)
 
     # have included this to differentiate between how different leagues handle ladders/tables
     league_tables <- season_stats_page %>% rvest::html_nodes("#content .table_wrapper") %>% rvest::html_elements("h2") %>% rvest::html_text()
 
     # we either want to detect the presence of League Table or Regular season and get the index of that to be able to extractg the table we want
-    if(length(grep("league table", tolower(league_tables))) > 0) {
-      league_tables_idx <- grep("league table", tolower(league_tables))
+    if(length(grep(competition_name, league_tables)) > 0) {
```
Comment (tonyelhabr) [inline R/get_season_team_stats.R:78]: [MLS](https://fbref.com/en/comps/22/Major-League-Soccer-Stats) is an ongoing league. We can see that "Regular Season" is still used, so we still need this if-else clause.

But for every other league that I checked, the competition name is now used for completed seasons.

**Review by JaseZiv — APPROVED** [review]
LGTM.

Thanks for addressing this one so quickly @tonyelhabr!

File: sd_meta.R
Comment (tonyelhabr) [inline sd_meta.R:1]: looks like an unintentionally committed file

**Comment by tonyelhabr** [comment]
@JaseZiv ah hold up, got one more code change i need to make 🤦

**Comment by tonyelhabr** [comment]
> @JaseZiv ah hold up, got one more code change i need to make 🤦

ok we're good now. needed to rename `season_url` to `single_season_url` in two places

**Comment by JaseZiv** [comment]
> > @JaseZiv ah hold up, got one more code change i need to make 🤦
> 
> ok we're good now. needed to rename `season_url` to `single_season_url` in two places

Happy for me to wait until the checks have run before merging @tonyelhabr?

**Comment by tonyelhabr** [comment]
> > > @JaseZiv ah hold up, got one more code change i need to make 🤦
> > 
> > 
> > ok we're good now. needed to rename `season_url` to `single_season_url` in two places
> 
> Awesome! Happy for me to wait until the checks have run before merging @tonyelhabr?

only failing tests are for unrelated functions (transfermarket stuff, `fb_league_stats()` due to RFC error). going to go ahead and merge to unblock @Paulj1989

**Pull request merged**